src/main/java/game/Submarine.java

```diff
@@ -149,28 +151,52 @@ public void executeStrategy() {
 			 * */
 
 				if (!map.enemyShips.isEmpty()) {
-
-					// TODO: a legmesszebbit kene ?
-					Vector2D enemyShip = new Vector2D(map.enemyShips.get(0).x(), map.enemyShips.get(0).y());
-
-					//current vector
-					Vector2D currentVector = new Vector2D(this.position.getX(), this.position.getY());
-
-					//target Vector
-					Vector2D targetVector2 = enemyShip.subtract(currentVector);
-
-					double targetVectorAngle2 = getAngleOfVector(targetVector2);
-					if (targetVectorAngle2 < 0) {
-						targetVectorAngle2 += 360;
-					}
-					log.info("Ship[{}]: TargetVectorAngle calculated: {}", this.id, targetVectorAngle2);
-
-					if (this.torpedoCooldown <= 0) {
-						this.actionQueue.add(Action.shoot(targetVectorAngle2));
-					}
+				
+				// TODO: a legmesszebbit kene ?
+
+				ProjectileLike enemy = map.enemyShips.get(0);
+				Vector2D enemyShip = new Vector2D(enemy.x(), enemy.y());
+				
+				//current vector
+				Vector2D currentVector = new Vector2D(this.position.getX(), this.position.getY());
+				
+				//target Vector
+				Vector2D targetVector2 = enemyShip.subtract(currentVector);
+
+				
+				double distanceFromTarget = currentVector.distance(targetVector2);
+				int numberOfRoundsToReachTarget = (int) Math.ceil(distanceFromTarget/this.map.mapConfig.torpedoSpeed);
+				
+				
+				// find out where enemy ship will be in numberOfRoundsToReachTarget rounds
+				XVector v = XVector.unit(enemy.rotation);
+				v = v.scale(enemy.speed);
+				
+//				// ha tul kozel van, mozogjon ellenkezo iranyba
+//				if(currentVector.distance(targetVector2) < map.mapConfig.torpedoExplosionRadius){
+//					this.actionQueue.add(Action.move(rotationDiff, map.mapConfig.maxAccelerationPerRound));
+////					conn.move(session.gameId, this.id, session.mapConfiguration.maxAccelerationPerRound, rotationDiff);
+//				}
+				
+				//hol lesz x kor mulva
+				for(int i=0; i<numberOfRoundsToReachTarget; i++){
+					targetVector2 = targetVector2.add(new Vector2D(v.x, v.y));
 				}
+				
+				double targetVectorAngle2 = getAngleOfVector(targetVector2);
+				if(targetVectorAngle2 < 0){
+					targetVectorAngle2 += 360;
+				}
+				log.info("Ship[{}]: TargetVectorAngle calculated: {}", this.id, targetVectorAngle2);
+				
+				if(canShootTorpedo()) {
+//					conn.shoot(map.gameId, this.id, targetVectorAngle2);
+					this.actionQueue.add(Action.shoot(targetVectorAngle2));
+					torpedosShotInRound = this.map.mapConfig.rounds;
```

Comment:
The fact that you put the torpedo shoot intent here does not mean that it's shot in the same round. We'll need to decide on action handling orders